sputnikdao2/src/policy.rs

```diff
@@ -237,8 +237,52 @@ impl VersionedPolicy {
 }
 
 impl Policy {
-    ///
-    /// Doesn't fail, because will be used on the finalization of the proposal.
+    pub fn add_or_update_role(&mut self, role: &RolePermission) {
+        for i in 0..self.roles.len() {
+            if &self.roles[i].name == &role.name {
+                env::log_str(&format!(
+                    "Updating existing role in the policy:{}",
+                    &role.name
+                ));
+                let _ = std::mem::replace(&mut self.roles[i], role.clone());
```

Comment:
interesting approach. I know in the distant future we'd like to write smart contracts that don't use `std` at all, but think it's fine here :)

Reply:
Interesting, I didn't know about that. Created an issue to try and address all the std use inside sputnik: https://github.com/near-daos/sputnik-dao-contract/issues/90